lib/suspenders/adapters/ci.rb

```diff
@@ -0,0 +1,87 @@
+module Suspenders
+  module Adapters
+    class Ci
+      def initialize(app_builder)
+        @app_builder = app_builder
+      end
+
+      def setup_ci(provider)
+        adapter_for(provider).configure
+      end
+
+      def configure_automatic_deployment(provider)
+        adapter_for(provider).configure_deployment
+      end
+
+      private
+
+      attr_reader :app_builder
+
+      def adapter_for(provider)
+        case provider
+        when 'travis' then Travis.new(app_builder)
+        when 'circle' then Circle.new(app_builder)
+        when 'codeship' then Codeship.new(app_builder)
+        end
+      end
+
+      class Base
```

Comment:
Would love feedback on this design -- I think it's a bit messy, but object inheritance seems like the most obvious way to do this